crates/kubelet/src/grpc_sock/windows/mod.rs

```diff
@@ -56,7 +67,17 @@ impl Stream for Socket {
                 return Poll::Pending;
             }
         };
-        return Poll::Ready(Some(UnixStream::new(stream)));
+        // The `new` function for a UnixStream is async so it can run on the
+        // right runtime version. All it needs is the handle, so there is no
+        // actual awaiting, so it is fine to loop here until the poll is ready
+        let mut fut = UnixStream::new(stream).boxed();
+        let stream = loop {
+            match fut.poll_unpin(cx) {
+                Poll::Ready(res) => break res,
+                Poll::Pending => continue,
+            }
+        };
+        Poll::Ready(Some(stream))
```

Comment:
This seems sketchy. I recommend changing your `new` function to be non-async and using [`now_or_never`](https://docs.rs/futures/0.3/futures/future/trait.FutureExt.html#method.now_or_never) in the constructor instead.

Reply:
This is most certainly sketchy! It is what made it work. I didn’t know about the now or never function, so I’ll give that a whirl